[tp] handle non-covered ops

feikou opened the pull request:
Summary: Only propagate sharding if the op has sharding strategy registered, otherwise mark in/out of the op as `Replicate`.

Test Plan: buck test mode/opt  -c fbcode.enable_gpu_sections=true //caffe2/test/distributed/_tensor/experimental:tp_transform

Differential Revision: D50747611

wanchaol reviewed (approved):
looks like a stacked diff output a PR that have duplicate content than the parent diff. for landing we can land the first one and then re-export this again to make sure github is happy

feikou committed:
[tp] handle non-covered ops (#112530)

Summary:

Only propagate sharding if the op has sharding strategy registered, otherwise mark in/out of the op as `Replicate`.

Test Plan: buck test mode/opt  -c fbcode.enable_gpu_sections=true //caffe2/test/distributed/_tensor/experimental:tp_transform

Reviewed By: wanchaol

Differential Revision: D50747611
